stellar-contract-env-host/src/host.rs

```diff
@@ -867,10 +870,54 @@ impl CheckedEnv for Host {
     }
 
     fn compute_hash_sha256(&self, x: Object) -> Result<Object, HostError> {
-        todo!()
+        let binary = self.visit_obj(x, |bin: &Vec<u8>| {
+            Ok(bin.clone()) //TODO:Why is this neccesary
```

Comment:
I think @sisuresh is referring to the `.clone()` here. I believe because this is returning a result from a local reference to Vec, you need to clone otherwise the lifetime checker will complain.

Reply:
And the `visit_obj` requires that the closure takes in an immutable reference.

Reply:
I removed this code